### PR TITLE
Fix plugin default detection

### LINKED ISSUE #2651: _set_by_cli spuriously flags all plugin variables to be set by cli
In new code intended to fix #2503, I use `_set_by_cli` as one criterion for deciding to save something in the renewal config file. I found that it sometimes returns `True`, resulting in saving items, even when I did not actually set those things from the CLI at all. The items in question that I observed so far were these:

```
apache_enmod = a2enmod
apache_server_root = /etc/apache2
apache_dismod = a2dismod
apache_handle_sites = True
apache_handle_modules = True
standalone_supported_challenges = "tls-sni-01,http-01"
apache_le_vhost_ext = -le-ssl.conf
apache_challenge_location = /etc/apache2
nginx_server_root = /etc/nginx
nginx_ctl = nginx
apache_vhost_root = /etc/apache2/sites-available
```

There could be others, but these were ones I saw so far.

Cc @pde

### PR DESCRIPTION
**Pull request by bmw**
Fixes #2651.

### CODE REVIEW

**Comment by bmw** [comment]
@jsha, `./bin/boulder-sa` exited early: [log1](https://gist.github.com/86c2096c408c715ccaee86d18af1f604), [log2](https://gist.github.com/238fe522ace44b629b18491eaa856655).

File: letsencrypt/cli.py [hunk 1]
```diff
@@ -87,6 +87,48 @@
...
+# These argparse parameters should be removed when detecting defaults.
+ARGPARSE_PARAMS_TO_REMOVE = ("const", "nargs", "type",)
+
+
+# These sets are used when to help detect options set by the user.
+EXIT_ACTIONS = set(("help", "version",))
+
+
+ZERO_ARG_ACTIONS = set(("store_const", "store_true",
+                        "store_false", "append_const", "count",))
```
Comment (pde) [inline letsencrypt/cli.py]: Stylistically, we should probably move all of this default detection machinery to a different file (though that will require inheriting from `HelpfulArgumentParser` rather than tweaking it) or at least a clearly marked section of this file, because it's probably confusing that these variables are scattered amongst the regular CLI parsing case.

But that can be another PR rather than this one.

File: letsencrypt/renewal.py [hunk 2]
```diff
@@ -102,16 +102,14 @@ def _restore_webroot_config(config, renewalparams):
     form.
     """
     if "webroot_map" in renewalparams:
-        # if the user does anything that would create a new webroot map on the
-        # CLI, don't use the old one
-        if not (cli.set_by_cli("webroot_map") or cli.set_by_cli("webroot_path")):
-            setattr(config.namespace, "webroot_map", renewalparams["webroot_map"])
+        if not cli.set_by_cli("webroot_map"):
```
Comment (pde) [inline letsencrypt/renewal.py]: I think the previous semantics might be less confusing to people here.  Suppose on day 1 you did:
```
letsencrypt certonly --webroot -w /var/www/ -d example.com,www.example.com
```
And then you did:
```
letsencrypt renew -w /var/www/moved
```
I believe the change you've made here would cause the webroot to remain `/var/www/` during renewal, because the client would construct and save a `webroot_map` during first execution.

Comment (pde) [inline letsencrypt/renewal.py]: Oh wait, I'm sorry, I see you've handled this case via `VAR_MODIFIERS`.

**Comment by pde** [comment]
LGTM!
